presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -138,9 +139,10 @@ private static InternalNode findCurrentNode(List<ServiceDescriptor> allServices,
         for (ServiceDescriptor service : allServices) {
             URI uri = getHttpUri(service, httpsRequired);
             OptionalInt thriftPort = getThriftServerPort(service);
+            InternalNodeSupply internalNodeSupply = getInternalNodeResource(service);
             NodeVersion nodeVersion = getNodeVersion(service);
             if (uri != null && nodeVersion != null) {
-                InternalNode node = new InternalNode(service.getNodeId(), uri, thriftPort, nodeVersion, isCoordinator(service));
+                InternalNode node = new InternalNode(service.getNodeId(), uri, thriftPort, nodeVersion, isCoordinator(service), Optional.of(internalNodeSupply));
```

Comment:
The optional.of(internalNodeSupply) seems redundant since inside internalNodeSupply, you already define generalPoolCapacityInBytes and numberOfCpuCores also as Optional